src/v/storage/compaction_reducers.cc

```diff
@@ -115,6 +116,11 @@ compacted_offset_list_reducer::operator()(compacted_index::entry&& e) {
 
 std::optional<model::record_batch>
 copy_data_segment_reducer::filter(model::record_batch&& batch) {
+    // do not compact batches other than raft data
+    if (batch.header().type != model::record_batch_type::raft_data) {
+        return std::move(batch);
```

Comment:
We need to compact other that raft data batches in consumer groups, at least:

- model::record_batch_type::group_commit_tx
- model::record_batch_type::tx_fence
- model::record_batch_type::group_prepare_tx
- model::record_batch_type::group_abort_tx

Maybe it's better to filter out not "not raft data" but only configuration batches?

Reply:
are those batches included in offset translation ?

Reply:
I've changed this to only include archival metadata and raft configuration.